.github/actions/fetch-codeql/action.yml

```
@@ -1,14 +1,22 @@
 name: Fetch CodeQL
 description: Fetches the latest version of CodeQL
+
+inputs:
+  channel:
+    description: 'The CodeQL channel to use'
+    required: false
+    default: 'nightly'
+
 runs:
   using: composite
   steps:
     - name: Fetch CodeQL
       shell: bash
+      env:
+        GITHUB_TOKEN: ${{ github.token }}
+        CHANNEL: ${{ inputs.channel }}
       run: |
         gh extension install github/gh-codeql
-        gh codeql set-channel nightly
+        gh codeql set-channel $CHANNEL
```

Some quotes in case there are funny characters in the input value.

```suggestion
        gh codeql set-channel "$CHANNEL"
```

---

things will probably crash if you don't use nightly/release, so it shouldn't be needed.  
But it's best practise, so yeah, I'll add the quotes.